src/atomate2/vasp/flows/adsorption.py

```diff
@@ -0,0 +1,223 @@
+"""Flow for calculating surface adsorption energies."""
+
+from __future__ import annotations
+
+from collections import defaultdict
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, Any
+
+from jobflow import Flow, Job, Maker
+
+from atomate2.vasp.jobs.adsorption import (
+    BulkRelaxMaker,
+    MolRelaxMaker,
+    MolStaticMaker,
+    SlabRelaxMaker,
+    SlabStaticMaker,
+    adsorption_calculations,
+    generate_adslabs,
+    generate_slab,
+    run_adslabs_job,
+)
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from pymatgen.core.structure import Molecule, Structure
+
+    from atomate2.forcefields.jobs import ForceFieldRelaxMaker, ForceFieldStaticMaker
+    from atomate2.vasp.jobs.base import BaseVaspMaker
+
+
+@dataclass
+class AdsorptionMaker(Maker):
+    """
+    Workflow that calculates the adsorption energy of a molecule on a surface.
+
+    The flow consists of the following steps:
+    1. Optimize the molecule structure and calculate its static energy.
+    2. Optimize the bulk structure.
+    3. Generate a slab structure using the optimized bulk structure and calculate its static energy.
+    4. Generate adsorption sites on the slab and calculate corresponding static energy.
+    5. Calculate the adsorption energy by calculating the energy difference between the slab with
+    adsorbed molecule and the sum of the slab without the adsorbed molecule and the molecule.
+
+    Parameters
+    ----------
+    name: str
+        Name of the flow.
+    bulk_relax_maker: BaseVaspMaker
+        Maker for bulk relaxation.
+    mol_relax_maker: BaseVaspMaker
+        Maker for molecule relaxation.
+    slab_relax_maker: BaseVaspMaker
+        Maker for slab relaxation with adsorption.
+    slab_static_maker: BaseVaspMaker
+        Maker for slab static energy calculation.
+    mol_static_maker: BaseVaspMaker
+        Maker for molecule static energy calculation.
+    """  # noqa: E501
+
+    name: str = "adsorption workflow"
+    mol_relax_maker: BaseVaspMaker | ForceFieldRelaxMaker | None = field(
+        default_factory=MolRelaxMaker
+    )
+
+    mol_static_maker: BaseVaspMaker | ForceFieldStaticMaker | None = field(
+        default_factory=MolStaticMaker
+    )
+
+    bulk_relax_maker: BaseVaspMaker | ForceFieldRelaxMaker | None = field(
+        default_factory=BulkRelaxMaker
+    )
+
+    slab_relax_maker: BaseVaspMaker | ForceFieldRelaxMaker | None = field(
+        default_factory=SlabRelaxMaker
+    )
+
+    slab_static_maker: BaseVaspMaker | ForceFieldStaticMaker | None = field(
+        default_factory=SlabStaticMaker
+    )
+
+    def make(
+        self,
+        molecule: Molecule,
+        structure: Structure,
+        min_vacuum: float = 20.0,
+        min_slab_size: float = 10.0,
+        min_lw: float = 10.0,
+        surface_idx: tuple[int, int, int] = (0, 0, 1),
```

Comment:
To be consistent with the other makers, can you put these arguments as dataclass fields instead?